### PR TITLE
integration-cli: fix swarm tests flakiness

### PR DESCRIPTION
**Pull request by tonistiigi**
relates to / addresses:

- https://github.com/moby/moby/issues/36903 Flaky test: TestAPISwarmServicesCreate
- https://github.com/moby/moby/issues/34988 Flaky test: TestAPISwarmRaftQuorum
- https://github.com/moby/moby/issues/37132 Flaky test: TestServicePlugin, TestCreateServiceSecretFileMode
- https://github.com/moby/moby/issues/32673 Flaky test: TestAPISwarmLeaderElection
- https://github.com/moby/moby/issues/28240 Flaky test: DockerSwarmSuite.TestSwarmRotateUnlockKey
- https://github.com/moby/moby/issues/38885 Flaky test: DockerSwarmSuite.TestSwarmClusterRotateUnlockKey
  - related fix for this test: https://github.com/moby/moby/pull/39616 Fix TestSwarmClusterRotateUnlockKey

SwarmKit bump:

full diff: https://github.com/docker/swarmkit/compare/fb584e7b501ec4683b5c3e62476d76b8a7e7d9f6...7dded76ec532741c1ad9736cd2bb6d6661f0a386

includes:

- docker/swarmkit#2808 Fix flaky tests
- docker/swarmkit#2866 Swap gometalinter for golangci-lint
- docker/swarmkit#2869 Increase max recv gRPC message size to initialize connection broker
- docker/swarmkit#2870 Fix update out of sequence

### CODE REVIEW

**Comment by thaJeztah** [comment]
`DockerSwarmSuite.TestAPISwarmRaftQuorum` failed on Janky and Experimental

`DockerSwarmSuite.TestAPISwarmServicesCreate` also failed on Experimental

**Comment by tonistiigi** [comment]
Pushed updates for: `TestAPISwarmLeaderElection` and `TestAPISwarmRaftQuorum`. `TestAPISwarmLeaderElection` now allows temporary context deadline exceeded and no-leader errors while the leader is switching. Ideally, these should never happen and we can create an issue to track if it is possible to guarantee that. The test still checks that leader election eventually happens and cluster state normalizes. For `TestAPISwarmRaftQuorum` I increased the timeout from default 30s like some other tests do as well. Probably for similar reasons as `TestAPISwarmLeaderElection` it sometimes gets deadline exceeded errors and when this happens it should have been already waited for 20sec, plus possible wait for heartbeats. If this doesn't help then temporarily I wouldn't mind allowing deadline exceeded error as well in addition to the specific error. More important part of the test is to check that the state normalizes.

For `TestAPISwarmServicesCreate` this seems actual swarm error with no workaround in the test. In the logs I see.
```
time="2019-07-15T22:22:17.270689016Z" level=error msg="failed transaction: update task desired state to REMOVE" error="update out of sequence" module=node node.id=puuje6xefqjhl5k42bfmqgh9v
time="2019-07-15T22:22:17.270715610Z" level=error msg="failed transaction: update task desired state to REMOVE" error="update out of sequence" module=node node.id=puuje6xefqjhl5k42bfmqgh9v
```
when it tries to remove a service.

**Comment by tonistiigi** [comment]
In experimental `TestServicePlugin` now failed from integration suite. The cause seems same as `TestAPISwarmServicesCreate` before.

```
time="2019-07-16T03:20:19.393117770Z" level=error msg="failed transaction: update task desired state to REMOVE" error="update out of sequence" module=node node.id=740tzgyr852ufnycchjw39kbp
time="2019-07-16T03:20:19.393133366Z" level=error msg="failed transaction: update task desired state to REMOVE" error="update out of sequence" module=node node.id=740tzgyr852ufnycchjw39kbp
time="2019-07-16T03:20:19.393148882Z" level=error msg="failed transaction: update task desired state to REMOVE" error="update out of sequence" module=node node.id=740tzgyr852ufnycchjw39kbp
```

If I understand correctly this is a completely internal error in swarm that causes the state change to be ignored with no way to control or get an error from the client.

All of these are covered in https://gist.github.com/tonistiigi/849f6551003fff2da43cbe0e21172e23

**Comment by dperny** [comment]
Opened a PR to fix the `Update out of sequence` error: https://github.com/docker/swarmkit/pull/2870

**Comment by tonistiigi** [comment]
Reverted memberlist change to check if it has an effect on tests or can be worked on separately.

**Review by andrewhsu — APPROVED** [review]
LGTM w.r.t. reducing the flakiness of tests

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by crosbymichael** [comment]
LGTM

File: integration-cli/docker_api_swarm_test.go [hunk 1]
```diff
@@ -313,13 +314,24 @@ func (s *DockerSwarmSuite) TestAPISwarmLeaderElection(c *check.C) {
...
+	var lastErr error
 	checkLeader := func(nodes ...*daemon.Daemon) checkF {
 		return func(c *check.C) (interface{}, check.CommentInterface) {
 			// clear these out before each run
 			leader = nil
 			followers = nil
 			for _, d := range nodes {
-				if d.GetNode(c, d.NodeID()).ManagerStatus.Leader {
+				n := d.GetNode(c, d.NodeID(), func(err error) bool {
+					if strings.Contains(errors.Cause(err).Error(), context.DeadlineExceeded.Error()) || strings.Contains(err.Error(), "swarm does not have a leader") {
```
Comment (thaJeztah) [inline integration-cli/docker_api_swarm_test.go]: Perhaps we should update this to take the same approach as https://github.com/moby/moby/pull/39168

**Pull request merged**